tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -847,6 +826,46 @@ impl LeaderTpuService {
                 }
             }
         }
+
+        Ok(())
+    }
+
+    async fn run_slot_watcher(
+        recent_slots: RecentLeaderSlots,
+        pubsub_client: Option<PubsubClient>,
+        exit: Arc<AtomicBool>,
+    ) -> Result<()> {
+        let Some(pubsub_client) = pubsub_client else {
+            return Ok(());
+        };
+
+        let (mut notifications, unsubscribe) = pubsub_client.slot_updates_subscribe().await?;
+
+        loop {
+            if exit.load(Ordering::Relaxed) {
+                break;
+            }
+
+            while let Ok(Some(update)) =
+                timeout(Duration::from_millis(10), notifications.next()).await
```

Comment:
Might be nice to create a const for this 10ms value and add a comment explaining why we chose it.

I'm assuming it's (1) long enough to make CPU consumption of this task minimal while being (2) short enough relative to slot times to provide timely information to subscribers